app/helpers/users_helper.rb

```diff
@@ -119,31 +119,66 @@ def authored_items(pseud, work_counts = {}, rec_counts = {})
     items.html_safe
   end
 
-  def log_item_action_name(action)
-    if action == ArchiveConfig.ACTION_ACTIVATE
-      t('users_helper.log_validated', default: 'Account Validated')
-    elsif action == ArchiveConfig.ACTION_ADD_ROLE
-      t('users_helper.log_role_added', default: 'Role Added: ')
-    elsif action == ArchiveConfig.ACTION_REMOVE_ROLE
-      t('users_helper.log_role_removed', default: 'Role Removed: ')
-    elsif action == ArchiveConfig.ACTION_SUSPEND
-      t('users_helper.log_suspended', default: 'Suspended until ')
-    elsif action == ArchiveConfig.ACTION_UNSUSPEND
-      t('users_helper.log_lift_suspension', default: 'Suspension Lifted')
-    elsif action == ArchiveConfig.ACTION_BAN
-      t('users_helper.log_ban', default: 'Suspended Permanently')
-    elsif action == ArchiveConfig.ACTION_WARN
-      t('users_helper.log_warn', default: 'Warned')
-    elsif action == ArchiveConfig.ACTION_RENAME
-      t('users_helper.log_rename', default: 'Username Changed')
-    elsif action == ArchiveConfig.ACTION_PASSWORD_RESET
-      t('users_helper.log_password_change', default: 'Password Changed')
-    elsif action == ArchiveConfig.ACTION_NEW_EMAIL
-      t('users_helper.log_email_change', default: 'Email Changed')
-    elsif action == ArchiveConfig.ACTION_TROUBLESHOOT
-      t('users_helper.log_troubleshot', default: 'Account Troubleshot')
-    elsif action == ArchiveConfig.ACTION_NOTE
-      t('users_helper.log_note', default: 'Note Added')
+  def log_item_action_name(item, user)
+    action = item.action
+
+    if [ArchiveConfig.ACTION_ADD_FNOK, ArchiveConfig.ACTION_REMOVE_FNOK].include?(action)
+      if item.fnok_user_id == user.id
+        if action == ArchiveConfig.ACTION_REMOVE_FNOK
+          return t(
+            "users_helper.log_removed_as_fnok",
+            default: "Removed as Fannish Next of Kin for: %{user_id}",
+            user_id: item.user_id
+          )
+        end
+
+        return t(
+          "users_helper.log_added_as_fnok",
+          default: "Added as Fannish Next of Kin for: %{user_id}",
+          user_id: item.user_id
+        )
+      end
+
+      if action == ArchiveConfig.ACTION_REMOVE_FNOK
+        return t(
+          "users_helper.log_remove_fnok",
+          default: "Fannish Next of Kin Removed: %{user_id}",
+          user_id: item.fnok_user_id
+        )
+      end
+
+      return t(
+        "users_helper.log_add_fnok",
+        default: "Fannish Next of Kin Added: %{user_id}",
+        user_id: item.fnok_user_id
+      )
+    end
```

Comment:
I think `log_item_action_name` is getting a little long with this. Could you move the logic for figuring out how to log FNOK actions into a separate method? 

It would also be a good idea to use our regular, `default`-less method of i18n here, even if we're leaving the `default` for the existing code.